sdk/identity/identity/src/util/tracing.ts

```diff
@@ -11,3 +14,45 @@ export const createSpan = createSpanFunction({
   packagePrefix: "Azure.Identity",
   namespace: "Microsoft.AAD"
 });
+
+/**
+ * From: https://github.com/Azure/azure-sdk-for-js/blob/46139daa3317a0d12e8b55b02b9d9cdf1b2e762a/sdk/appconfiguration/app-configuration/src/internal/tracingHelpers.ts
+ * Traces an operation and properly handles reporting start, end and errors for a given span
+ *
+ * @param operationName - Name of a method in the TClient type
+ * @param options - An options class, typically derived from \@azure/core-http/RequestOptionsBase
+ * @param fn - The function to call with an options class that properly propagates the span context
+ *
+ * @internal
+ */
+export async function trace<ReturnT>(
```

Comment:
this will get affected by Richard's PR #14208

Reply:
Richard is aware that we'll have merge conflicts.